pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/groupby/DictionaryBasedGroupKeyGenerator.java

```diff
@@ -612,7 +656,8 @@ public void processMultiValue(int numDocs, int[][] outGroupIds) {
     private int getGroupId(long rawKey) {
       int numGroups = _groupIdMap.size();
       if (numGroups < _globalGroupIdUpperBound) {
-        return _groupIdMap.computeIfAbsent(rawKey, k -> numGroups);
+        int id = _groupIdMap.putIfAbsent(rawKey, numGroups);
+        return id == _groupIdMap.defaultReturnValue() ? numGroups : id;
```

Comment:
Might be slightly faster to do
```suggestion
        return id == INVALID_ID ? numGroups : id;
```

Reply:
Maybe, the intent here was just to remove the capturing lambda while I was looking at something nearby. There’s no justification for this change, except to eliminate an allocation.